File: python/assistant-on-fastapi/app/errors/error_code.py
from enum import Enum

from fastapi import status


class ErrorCode(Enum):
    UNKNOWN = -1
    DB_FAILED = 1
    UNSUPPORTED_ML_MODEL = 2


def to_http_status_code(error_code: ErrorCode):
    error_code_status_map = {
        ErrorCode.UNSUPPORTED_ML_MODEL: status.HTTP_400_BAD_REQUEST
    }
    return error_code_status_map.get(error_code, status.HTTP_500_INTERNAL_SERVER_ERROR)

File: ts-backend/assistant-on-express/src/errors/error-code.ts
import { StatusCodes } from "http-status-codes"

enum ErrorCode {
  UNKNOWN = -1,
  MISTRAL_AI_INTEGRATION_FAILED = 1,
  MISTRAL_AI_INTEGRATION_MISCONFIGURED = 2,
}

export default ErrorCode

export function toHttpStatusCode(errorCode: ErrorCode): number {
  const errorCodeStatusMap: Record<ErrorCode, number> = {
    [ErrorCode.UNKNOWN]: StatusCodes.INTERNAL_SERVER_ERROR,
    [ErrorCode.MISTRAL_AI_INTEGRATION_FAILED]: StatusCodes.INTERNAL_SERVER_ERROR,
    [ErrorCode.MISTRAL_AI_INTEGRATION_MISCONFIGURED]: StatusCodes.INTERNAL_SERVER_ERROR,
  }

  return errorCodeStatusMap[errorCode] ?? StatusCodes.INTERNAL_SERVER_ERROR
}
